### PR TITLE
Bug: Reconnect pane not appearing when reconnect is configured 

### PR DESCRIPTION
**Pull request by sarojkpr**
- Fixing reconnect pane is not appearing in chat widget even when reconnect is configured. 
- Fixing customer conversation forcefully ended after reconnect time due to first issue.

Screenshots:
**Auth chat (with reconnect prompt)**
   _Previous session detected:_
image

  _Continue Conversation:_
image

  _Start new conversation:_
image

 _Customer chat not forcefully ended by OC:_
image

**Auth chat (with reconnect id)**
image

**Unauth Chat (with reconnect id)**
image
  
  _Customer chat not ended forcefully:_
image

Sample html to test: 
Unauth: https://sacadlcw001.blob.core.windows.net/lcwtpc/test2_reconnect_unauth.html
Auth: https://sacadlcw001.blob.core.windows.net/lcwtpc/test2_reconnect_auth.html
Script: https://sacadlcw001.blob.core.windows.net/lcwtpc/reconnectpane/v2scripts/LiveChatBootstrapper.js

### CODE REVIEW

File: chat-widget/src/components/livechatwidget/common/startChat.ts [hunk 1]
```diff
@@ -268,12 +269,15 @@ const setCustomContextParams = (chatSDK: any) => {
...
+    const requestId = state.domainStates?.liveChatContext?.requestId;
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     let conversationDetails: any = undefined;
 
     //Preserve old requestId
     const oldRequestId = chatSDK.requestId;
+    dispatch({ type: LiveChatWidgetActionType.SET_INITIAL_CHAT_SDK_REQUEST_ID, payload: oldRequestId });
+
     try {
         chatSDK.requestId = requestId;
```
Comment (nareshme2016) [inline chat-widget/src/components/livechatwidget/common/startChat.ts:282]: What if this is null

**Review by nareshme2016 — APPROVED** [review]
Approved with comment.

File: chat-widget/src/components/reconnectchatpanestateful/ReconnectChatPaneStateful.tsx [hunk 2]
```diff
@@ -27,6 +27,7 @@ export const ReconnectChatPaneStateful = (props: IReconnectChatPaneStatefulParam
             await initStartChat(optionalParams);
         } else {
             dispatch({ type: LiveChatWidgetActionType.SET_RECONNECT_ID, payload: undefined });
+            chatSDK.requestId = state.domainStates.initialChatSdkRequestId;
```
Comment (abhinab22mohanty) [inline chat-widget/src/components/reconnectchatpanestateful/ReconnectChatPaneStateful.tsx:30]: We might need to handle null case here as mentioned

File: CHANGE_LOG.md [hunk 3]
```diff
@@ -104,3 +104,4 @@ All notable changes to this project will be documented in this file.## [Unreleas
 - Fixed Postchat loading twice issue
 - Adding an event listener in case the chat window is hidden for notifications
 - Adding support for bot survey
+- Fixing reconnect pane not appearing
```
Comment (abhinab22mohanty) [inline CHANGE_LOG.md:107]: We might need to add more details to issues fixed in this PR

Fixing reconnect pane is not appearing in chat widget even when reconnect is configured.
Fixing customer conversation forcefully ended after reconnect time due to first issue.

File: chat-widget/src/components/livechatwidget/common/reconnectChatHelper.ts [hunk 4]
```diff
@@ -107,10 +107,12 @@ const redirectPage = (redirectURL: string, redirectInSameWindow: boolean) => {
...
 };
 
-const isReconnectEnabled = (chatConfig: ChatConfig) => {
-    return chatConfig &&
-        chatConfig.LiveWSAndLiveChatEngJoin?.msdyn_enablechatreconnect &&
-        (chatConfig.LiveWSAndLiveChatEngJoin.msdyn_enablechatreconnect === "true" || chatConfig.LiveWSAndLiveChatEngJoin.msdyn_enablechatreconnect === true);
+const isReconnectEnabled = (chatConfig?: ChatConfig): boolean => {
+    if (chatConfig) {
+        const reconnectEnabled = (chatConfig.LiveWSAndLiveChatEngJoin?.msdyn_enablechatreconnect?.toLowerCase() === "true");
+        return reconnectEnabled;
```
Comment (akagar-microsoft) [inline chat-widget/src/components/livechatwidget/common/reconnectChatHelper.ts:113]: nit: return chatConfig.LiveWSAndLiveChatEngJoin?.msdyn_enablechatreconnect?.toLowerCase() === "true";

File: CHANGE_LOG.md (hunk 3 above)
Comment (sarojkpr) [reply CHANGE_LOG.md]: Updated a bit, we definitely don't want to keep the change.log to have PR descriptions.

File: chat-widget/src/components/livechatwidget/common/reconnectChatHelper.ts (hunk 4 above)
Comment (sarojkpr) [reply chat-widget/src/components/livechatwidget/common/reconnectChatHelper.ts:113]: This is intentional due to the complexity of .(dot)s here for improve readability of code.

**Review by marghayk — APPROVED** [review]

**Pull request merged**